state/protocol/convert.go

```diff
@@ -74,20 +75,45 @@ func ToEpochCommit(epoch Epoch) (*flow.EpochCommit, error) {
 	if err != nil {
 		return nil, fmt.Errorf("could not get epoch dkg: %w", err)
 	}
-	dkgParticipants, err := ToDKGParticipantLookup(dkg, participants.Filter(filter.HasRole(flow.RoleConsensus)))
+	dkgParticipantKeys, err := GetDKGParticipantKeys(dkg, participants.Filter(filter.IsValidDKGParticipant))
 	if err != nil {
-		return nil, fmt.Errorf("could not compute dkg participant lookup: %w", err)
+		return nil, fmt.Errorf("could not get dkg participant keys: %w", err)
 	}
 
 	commit := &flow.EpochCommit{
-		Counter:         counter,
-		ClusterQCs:      qcs,
-		DKGGroupKey:     dkg.GroupKey(),
-		DKGParticipants: dkgParticipants,
+		Counter:            counter,
+		ClusterQCs:         qcs,
+		DKGGroupKey:        dkg.GroupKey(),
+		DKGParticipantKeys: dkgParticipantKeys,
 	}
 	return commit, nil
 }
 
+// GetDKGParticipantKeys retrieves the canonically ordered list of DKG
+// participant keys from the DKG.
+func GetDKGParticipantKeys(dkg DKG, participants flow.IdentityList) ([]crypto.PublicKey, error) {
+
+	keys := make([]crypto.PublicKey, 0, len(participants))
+	for i, identity := range participants {
+
+		index, err := dkg.Index(identity.NodeID)
+		if err != nil {
+			return nil, fmt.Errorf("could not get index (node=%x): %w", identity.NodeID, err)
+		}
+		key, err := dkg.KeyShare(identity.NodeID)
+		if err != nil {
+			return nil, fmt.Errorf("could not get key share (node=%x): %w", identity.NodeID, err)
+		}
+		if uint(i) != index {
```

Comment:
It would be difficult to have this sanity check in the filter function, since it is a simple stateless function. Plus, all we could do if the sanity check failed there would be to panic.